lookout/style/reporter.py

```diff
@@ -0,0 +1,146 @@
+"""General utilities to generate performance reports for analyzers."""
+
+import logging
+import os
+import shutil
+import tempfile
+from typing import Any, Dict, Iterable, Iterator, NamedTuple, Optional, Sequence, Tuple
+
+from lookout.core.analyzer import Analyzer
+from lookout.core.helpers.analyzer_context_manager import AnalyzerContextManager
+
+
+class Reporter:
+    """
+    Base class to create performance reports for the analyzer.
+
+    To create a reporter for yours Analyzer you should inherit two classes.
+    1. Inherit SpyAnalyzer from Analyzer you want to estimate. SpyAnalyzer `analyze` function
+       should be overwritten to return all information you need for the next performance analysis
+       in the Comments with a JSON object. See `TyposAnalyzerSpy` as an example.
+    2. Inherit MyReporter from this Reporter class. It is expected that you have a dataset that
+       you feed to `Reprter.run()` method. After that dataset rows are passed to
+       `_trigger_review_event` to trigger Your analyzer analyze method and send a result to
+       `_generate_reports` method. If you need to summarize your reports or make a reduced report
+       override `_finalize` method.
+
+       Note, that if you want to create several reports for the data
+       (like test, train reports for example) you should properly override both
+       `get_report_names()` and `_generate_reports()` functions.
+    """
+
+    _log = logging.getLogger("Reporter")
+
+    # Reporter can work only with analyzers that provides json in output message.
```

Comment:
```suggestion
    # Reporter can work only with analyzers that provide json in output message.
```
^still did not get this comment tbh

Reply:
Well, it is not true actually. Since I describe everything well in the docstring let's remove the comment.